### PR TITLE
Added Number of Divisors in CPP

### LINKED ISSUE #4732: Number of Divisors of a Number
## 💥 Proposal

(A clear and concise description of what the proposal is.)

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

YES.

### PR DESCRIPTION
**Pull request by RaiyanMahin**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES.

### Description

Given a Number. Find it's Number of Divisors.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #4732

### CODE REVIEW

File: C-Plus-Plus/math/number_of_divisors.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    {
+        if(Number % i == 0)
+        {
+            /* Here i is a divisor of that number
+            ( Number / i ) is also a divisor
+            Ex : 16
+            16 / 2 == 8
+            8 , 2 both are divisor of 16
+            so, we will increment the number_of_divisors by 2
+            */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/number_of_divisors.cpp]: Remove the redundant comments

File: C-Plus-Plus/math/number_of_divisors.cpp [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            number_of_divisors += 2;
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/number_of_divisors.cpp]: Remove the space

File: C-Plus-Plus/math/number_of_divisors.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        }
+    }
+    if(Number % limit == 0)
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/number_of_divisors.cpp:29]: Add the new line above the if condition

File: C-Plus-Plus/math/number_of_divisors.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+int main()
+{
+        cout << "Enter the number : \n";
+	int Number;
+	cin >> Number;
+	int number_of_divisors_of_the_number = get_number_of_divisors_of_the_number(Number);
+	cout << "Number of Divisors of this Number is : ";
+	cout << number_of_divisors_of_the_number << endl;
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/number_of_divisors.cpp]: Align your code properly also `number_of_divisors_of_the_number` rename this to `number_of_divisors`

File: C-Plus-Plus/math/number_of_divisors.cpp [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        }
+    }
+    if(Number % limit == 0)
+    {
+        if(limit * limit == Number)
+        {
+            /* means perfect square number
+            Ex : 16
+            4 * 4 == 16
+            */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/number_of_divisors.cpp]: Remove the redundant comments

File: C-Plus-Plus/math/number_of_divisors.cpp [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            number_of_divisors++;
+        }
+        else
+        {
+            /* Here i is a divisor of that number
+            ( Number / i ) is also a divisor
+            so, we will increment the number_of_divisors by 2
+            */
+            number_of_divisors += 2;
+
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/math/number_of_divisors.cpp]: Remove the redundant comments and space

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by RaiyanMahin** [comment]
all done!

**Review by ankitaggarwal23 — APPROVED** [review]

**Comment by HarshCasper** [comment]
Please make an issue here and raise a PR, for an article as part of our Docs repository: https://github.com/TesseractCoding/NeoAlgo-Docs

It would be much appreciated. Thanks!

**Pull request merged**